Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -1322,26 +1329,54 @@ public static Texture2D DownloadImage(string url)
                 }
             }
 
-            if (!timedout && !string.IsNullOrEmpty(request.error))
-            {
-                LogVerbose(request.error);
-            }
-
             Texture2D result = null;
 
-            if (!timedout && string.IsNullOrEmpty(request.error))
-            {
-                result = request.texture;
+            if (timedout) {
+                LogVerbose("Downloading image {0} timed out! Took more than 750ms.", url);
+            }
+            else {
+                if (string.IsNullOrEmpty(request.error)) {
+                    result = request.textureNonReadable;
+                    LogVerbose("Downloading image {0} took {1} ms", url, stopwatch.ElapsedMilliseconds);
+                }
+                else
+                    LogVerbose("Request error: " + request.error);
             }
 
-            LogVerbose(
-                timedout ? "Downloading image {0} timed out! Took more than 750ms." : "Downloading image {0} took {1} ms",
-                url,
-                stopwatch.ElapsedMilliseconds);
+           
+            if (result != null && !fromCache) {
+                CacheTextureOnDisk(url, request.bytes);
+            }
 
+            request.Dispose();
             return result;
         }
 
+        private static void CacheTextureOnDisk(string url, byte[] bytes) {
+            string diskPath = GetFilePath(url);
+            File.WriteAllBytes(diskPath, bytes);
+        }
+
+        private static bool ExistsInDiskCache(string url) {
+            return File.Exists(GetFilePath(url));
+        }
+
+        private static string GetFilePath(string url) {
+            return Path.Combine(Application.temporaryCachePath, GetHash(url));
+        }
+
+        private static string GetHash(string s) {
+            if (string.IsNullOrEmpty(s))
+                return null;
+            MD5CryptoServiceProvider md5 = new MD5CryptoServiceProvider();
```

Comment:
Would love if the cache entries had an expiry associated with them so that server updates would eventually be captured.

Reply:
Or you might try a HEAD query against the Url to compare the Entity properties.